src/global/pidfile.cc

```diff
@@ -96,3 +96,27 @@ int pidfile_remove(void)
   pid_file[0] = '\0';
   return 0;
 }
+
+int test_pid_file_in_use(const md_config_t *conf) 
+{
+  int fd;
+  if (conf->pid_file.empty()) {
+    return pidfile_remove();
+  }
+
+  fd = ::open(conf->pid_file.c_str(), O_CREAT|O_WRONLY, 0644);
```

Comment:
yeah, fd is leaked on purpose, because pid-file needs to keep open whole lifecycle, and if let it be leaked, a global varible can be omitted.